test/system/group_offer_categories_test.rb

```diff
@@ -3,11 +3,14 @@
 class GroupOfferCategoriesTest < ApplicationSystemTestCase
   def setup
     @user = create :user
-    login_as @user
+    @department = create :department
+    @department_manager = create :user, :with_departments
+    @group_offer_category = create :group_offer_category, category_name: 'Schwimmen'
   end
 
 
-  test 'create new group offer category and update it' do
+  test 'superadmin can create new group offer' do
```

Comment:
Since these tests are pretty much the same in pairs, please check if you can iterate over the users and make the 4 tests 2. In that case printing a text at the beginning of each test with the logged in user would be helpful. Otherwise checking only for the superadmin should be enough since the ability to create or update a category is solved through policies.

Reply:
Instead of looping through the users inside the tests you can also define the tests dynamically *inside* the loop :)

```ruby
users = {
  'superadmin' => create(:user),
  'department manager' => create(:user, :with_departments),
}

users.each do |label, user|
  test "#{label} can update a category" do
    login_as user
    ...
  end

  test "#{label} can create new group offer" do
    login_as user
    ...
  end
end
```

This will give you the same 4 tests with correct error messages etc.

Reply:
Nice! One small comment though for the implementation, :with_departments doesn't change the user role, so role: 'department_manager' should be added, as well.

Reply:
@toupeira: chrysa and me tried to set the up the tests like this, but couldn't get it working. I decided to simply reduce the tests, since we check the users already in the model test.